### ISSUE TITLE
Increasing a Variable Separately For Each Player

### ISSUE THREAD

**Issue by jessaenn** [haxball/haxball-issues#1184]
As the title says, does anyone have any idea about how to increase a variable separately for each player? I’ve tried couple of ways by creating an array which holds the ID and name datas of players but they didn’t be the solution that I’m looking for. If you have something sensible, you can show with your short own code as an example. I hope someone has.

**Comment by dapucita** [comment]
Hello. I'm newbie for coding, but I think I can show you an solution.

in my project Haxbotron, I wrote my bot in Typescript.

I used `Interface` for implement basic player data `PlayerObject`, and extend it when other custom variables are need.

Basic PlayerObject Model
https://github.com/dapucita/haxbotron/blob/master/model/PlayerObject.ts

Extended Player Model
https://github.com/dapucita/haxbotron/blob/master/model/Player.ts
`class Player implements PlayerObject`


So when new player has joined, my bot create new player object from `Player` class and allocate it into `Map` object.

**Comment by Raamyy** [comment]
```js
let playersObject = {}

function incrementPlayer(player){
    playersObject[player.id] = playersObject[player.id] || 0; // If the playersObject[player.id] has a value leave it as it is, else set it to zero.
    playersObject[player.id]++;
}
```

The above code holds `playerObject` data structure that it's **key** is the `player.id`, and it's **value** is a counter which is initially set to zero and is incremented using  `incrementPlayer()` function.

**Comment by jessaenn** [comment]
Firstly, thanks for your responses! 

I got the main point of two of your examples very well but the one that @Raamyy has shared seems more appropriate for my purpose. I’ve fixed my problem. 

Before opening this issue, the below code was the latest way I chose to try to do it. I thought I had to create two variables which one of them’s value is 0 by equalizing both each other.

`let playersData = {};`
`var increaseThis = 0;`

`increaseThis == playersData[player.id];`

`increaseThis[player.id]++;`
